### DISCUSSION TITLE
keycloak deployment on k8s mode production bug Resource not found

### DISCUSSION THREAD

**Discussion by tienlq2011** [keycloak/keycloak#33149] [Archive]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

core

### Describe the bug

Hello everyone, I have deployed Keycloak on Kubernetes and am running it in production mode. Everything was going well with the logs as shown below, but I encountered an issue when I set up a domain configured through an Envoy proxy to forward to Keycloak. Here are my deployment and service files:
deployment file :
apiVersion: apps/v1
kind: Deployment
metadata:
  name: lzt-id
  namespace: zerotrust
spec:
  replicas: 1
  selector:
    matchLabels:
      app: lzt-id
  template:
    metadata:
      labels:
        app: lzt-id
    spec:
      initContainers:
      - name: init-permissions
        image: busybox:1.36.1
        command: ['sh', '-c', 'chmod 666 /opt/keycloak/lib/quarkus/transformed-bytecode.jar || true']
        volumeMounts:
        - name: keycloak-config-volume
          mountPath: /opt/keycloak/
      containers:
      - name: lzt-id
        image: quay.io/keycloak/keycloak:25.0.2
        securityContext:
          runAsUser: 0
        volumeMounts:
        - name: keycloak-config-volume
          mountPath: /opt/keycloak/
        - name: keycloak-tls
          mountPath: /etc/x509/https
          readOnly: true
        env:
        - name: DB_VENDOR
          value: ${DB_VENDOR}
        - name: DB_ADDR
          value: ${DB_ADDR}
        - name: DB_DATABASE
          value: ${DB_DATABASE}
        - name: DB_USER
          value: ${DB_USER}
        - name: DB_PASSWORD
          value: ${DB_PASSWORD}
        - name: KEYCLOAK_HOSTNAME
          value: id.lab.linksafe.vn
        - name: KEYCLOAK_FRONTEND_URL
          value: ${KEYCLOAK_FRONTEND_URL}
        - name: PROXY_ADDRESS_FORWARDING
          value: "true"
        - name: KC_HTTP_RELATIVE_PATH
          value: /id/
        - name: KEYCLOAK_ADMIN
          value: admin
        - name: KEYCLOAK_ADMIN_PASSWORD
          value: admin
        command: ["/bin/sh", "-c"]
        args: ["/opt/keycloak/bin/kc.sh build && /opt/keycloak/bin/kc.sh start --proxy-headers xforwarded --hostname=id.lab.linksafe.vn --https-certificate-file=/etc/x509/https/tls.crt --https-certificate-key-file=/etc/x509/https/tls.key"] 
      volumes:
      - name: keycloak-config-volume
        hostPath:
          path: /root/deploy-zerotrust/setup-hostpath/keycloak
          type: Directory
      - name: keycloak-tls
        secret:
          secretName: keycloak-tls-secret
service file:
apiVersion: v1
kind: Service
metadata:
  name: lzt-id
  namespace: zerotrust
spec:
  type: NodePort
  selector:
    app: lzt-id
  ports:
    - name: http
      protocol: TCP
      port: 8080
      targetPort: 8080
      nodePort: 31111
    - name: https
      protocol: TCP
      port: 8443
      targetPort: 8443
      nodePort: 31112
bug : 
![image](https://github.com/user-attachments/assets/c4e600d8-a892-46e0-b61c-6bf4359ca488)
 
![image](https://github.com/user-attachments/assets/763289e6-edff-4833-b03b-d59086f6be95)

### Version

25.0.2

### Regression

- [X] The issue is a regression

### Expected behavior

bug:
![image](https://github.com/user-attachments/assets/e970d83a-02c0-404a-9204-15c5c61f1979)
![image](https://github.com/user-attachments/assets/ba66a33e-f2cc-4a6a-8000-dbea874eee67)

### Actual behavior

keycloak deploy on k8s 

### How to Reproduce?

keycloak deploy on k8s

**Comment by shawkins** [comment]
@tienlq2011 you marked as a regression, why?

Also it might to know what your env variables are doing - with the exception of the admin username and password (this is changing in KC 26) the env variable used by keycloak start with KC_ 

Converting to a discussion until there's more clarity about if there's anything wrong here, or if this is just a general configuration question.
